Router

marco9446 opened the pull request:
Create router
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/triglian%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23issuecomment-227730222%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-06-22T12%3A39%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/behaviors/qea-redux-store.html%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881752%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22the%20initial%20state%20has%20a%20value%3F%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A39%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/behaviors/qea-redux-store.html%3AL74-81%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%2058%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882098%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22camelcase%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A42%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%3AL4-53%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html%2093%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881805%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22camelcase%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A40%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html%3AL1-297%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html%2070%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881771%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22camelcase%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A39%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html%3AL1-297%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%2037%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882105%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22camelcase%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A42%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%3AL4-53%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/qea-canvas/qea-canvas.html%2058%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66883083%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22nodesObjects%20-%3E%20nodeObjects%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A50%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/qea-canvas/qea-canvas.html%3AL29-58%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882065%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22multy-%3Emulti%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A42%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html%3AL4-53%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/qea-canvas/qea-canvas.html%2098%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66883203%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22nodeObjects%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A51%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/qea-canvas/qea-canvas.html%3AL73-134%22%7D%2C%20%22Pull%205b6f209f03e7ca20eefc44999afbbb80a54bcad5%20app/elements/asq-highlight-q-editor/asq-highlight-q-editor-styles.html%2013%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882004%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22tag%20is%20open%22%2C%20%22created_at%22%3A%20%222016-06-13T22%3A41%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/573280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/triglian%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20app/elements/asq-highlight-q-editor/asq-highlight-q-editor-styles.html%3AL51-55%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23issuecomment-227730222%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881752%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881771%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66881805%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882004%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882065%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882098%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66882105%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66883083%22%2C%20%22https%3A//github.com/ASQ-USI/question-editor-asq/pull/18%23discussion_r66883203%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>

<img src='http://www.codereviewhub.com/site/github-approved-avatar.png'><a href='https://github.com/triglian'><img src='https://avatars.githubusercontent.com/u/573280?v=3' width=34 height=34></a>
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/behaviors/qea-redux-store.html 5'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66881752'>File: app/behaviors/qea-redux-store.html:L74-81</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> the initial state has a value?
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html 70'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66881771'>File: app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html:L1-297</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> camelcase
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html 93'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66881805'>File: app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html:L1-297</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> camelcase
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-highlight-q-editor/asq-highlight-q-editor-styles.html 13'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66882004'>File: app/elements/asq-highlight-q-editor/asq-highlight-q-editor-styles.html:L51-55</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> tag is open
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html 19'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66882065'>File: app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:L4-53</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> multy->multi
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html 58'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66882098'>File: app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:L4-53</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> camelcase
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html 37'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66882105'>File: app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:L4-53</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> camelcase
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/qea-canvas/qea-canvas.html 58'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66883083'>File: app/elements/qea-canvas/qea-canvas.html:L29-58</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> nodesObjects -> nodeObjects
- [ ] <a href='#crh-comment-Pull 5b6f209f03e7ca20eefc44999afbbb80a54bcad5 app/elements/qea-canvas/qea-canvas.html 98'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ASQ-USI/question-editor-asq/pull/18#discussion_r66883203'>File: app/elements/qea-canvas/qea-canvas.html:L73-134</a></b>
- <a href='https://github.com/triglian'><img border=0 src='https://avatars.githubusercontent.com/u/573280?v=3' height=16 width=16'></a> nodeObjects

<a href='https://www.codereviewhub.com/ASQ-USI/question-editor-asq/pull/18?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/ASQ-USI/question-editor-asq/pull/18?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/ASQ-USI/question-editor-asq/pull/18?approve=0'><img src='http://www.codereviewhub.com/site/github-undo-approve.png' height=26></a>&nbsp;<a href='https://github.com/ASQ-USI/question-editor-asq/pull/18'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

marco9446 committed:
create a minimal editor for asq-buckets-q and made some minor fixes

marco9446 committed:
fix text area in canvas

marco9446 committed:
improve user notification and create a code logger for canvas

marco9446 committed:
add dialog to check generated code

marco9446 committed:
add cancel bt

marco9446 committed:
fix restore state when edit question

marco9446 committed:
temporary fix for asq-buckets-q

marco9446 committed:
create test for qea elements

marco9446 committed:
create testS for the questions editors

marco9446 committed:
create a basic router

marco9446 committed:
create a basic router

marco9446 committed:
create a basic router

marco9446 committed:
create a basic router

marco9446 committed:
add the ability to clone questions

marco9446 committed:
add the ability to clone questions

marco9446 committed:
add the ability to clone questions

marco9446 committed:
add the ability to clone questions

marco9446 committed:
fix path and lint complaints

marco9446 committed:
Merge branch 'router' of https://github.com/ASQ-USI/question-editor-asq into router

marco9446 committed:
fix creation label issue in asq-buckets-q-editor

marco9446 committed:
code revision

marco9446 committed:
fix wrong import

marco9446 committed:
remove the necessity of # in URL for local files

marco9446 committed:
implement the possibility to save and restore an exercise

marco9446 committed:
improve user experience

marco9446 committed:
include all the questions in the same exercise

marco9446 committed:
make some style changes

marco9446 committed:
change import strategy and spinner css

marco9446 committed:
change router implementation

BUG:
- setting path from main app, requires  async
- timing issues in qea-editor-page

triglian committed:
merge with remote

triglian committed:
add start command

triglian committed:
fix typo

marco9446 committed:
add source map to babel

triglian committed:
move subrouters to their elements and debounce handling question editor route

marco9446 committed:
improved build process,  improved router navigation, add test

marco9446 committed:
fix build process

marco9446 committed:
fix restore properties

marco9446 committed:
fix error during properties restore

marco9446 committed:
fix typos and covert to lowercase all HTML attributes

triglian commented on app/behaviors/qea-redux-store.html:
```diff
@@ -74,7 +74,7 @@
       });
     case 'RESET_STATE':
-      state = undefined; // eslint-disable-line no-param-reassign
+      state = { id: 1234, elements: action.state || [] }; // eslint-disable-line no-param-reassign
```
the initial state has a value?

triglian commented on app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html:
```diff
@@ -0,0 +1,296 @@
+        bucket list
+      </div>
+      <qea-items-list
+        new-Item-Label="Insert new buckets"
```
camelcase

triglian commented on app/elements/asq-buckets-q-editor/asq-buckets-q-editor.html:
```diff
@@ -0,0 +1,296 @@
+      <qea-property-editor-repeater id="qeaRepeatValues"
+        property-Array=[[questionProps]]
+        values={{qeaRepeatValues}}
+        is-Advanced-User=[[viewAdvanced]]>
```
camelcase

triglian commented on app/elements/asq-highlight-q-editor/asq-highlight-q-editor-styles.html:
```diff
@@ -50,4 +51,4 @@
     </style>
   </template>
-</dom-module>
+</dom-module
```
tag is open

triglian commented on app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:
```diff
@@ -4,50 +4,49 @@
     <style include="qea-three-column-layout"></style>
     <style>
-
+      .multyChoiceOption{
```
multy->multi

triglian commented on app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:
```diff
@@ -4,50 +4,49 @@
-        value={{_optionsValue}}>
-      </qea-options-editor>
+      <qea-items-list
+        new-Item-Label="Insert an option"
```
camelcase

triglian commented on app/elements/asq-multi-choice-q-editor/asq-multi-choice-q-editor.html:
```diff
@@ -4,50 +4,49 @@
         name='title'
         label="question Title"
         type='string'
+        widget-Type='text-code'
```
camelcase

triglian commented on app/elements/qea-canvas/qea-canvas.html:
```diff
@@ -23,10 +29,29 @@
     <div id="canvas" class="flex" on-tap="_selectItem" on-mouseover="_onHovered" on-mouseout="_onUnhovered">
       <sortable-js handle=".handle" scroll  animation="200" force-Fallback="true" >
-        <template is="dom-repeat" items={{objs}}  id="repeater">
+        <template is="dom-repeat" items={{nodesObjects}}  id="repeater">
```
nodesObjects -> nodeObjects

triglian commented on app/elements/qea-canvas/qea-canvas.html:
```diff
@@ -48,36 +73,61 @@
+          * Questions or text to be rendered in canvas. This Prop is set by the redux store
+          * @type {[{id:String, type:String, html:String, props: Object}, ...]}
+          */
+          nodesObjects: {
```
nodeObjects

marco9446 committed:
Merge branch 'buckets-editor' into router

triglian committed:
fix typo

triglian committed:
Add mountpath and associated machinery
 - gulp task now generates index.html with base and mountpath configured
 - add .env file to read the mount path
 - add .env to gitinore
 - add sample .env.example file

triglian committed:
add .env.example

triglian commented:
:shipit:
